index.d.ts

```diff
@@ -0,0 +1,93 @@
+/// <reference types="node"/>
+import {Hash} from 'crypto';
+import {Readable as ReadableStream} from 'stream';
+
+export type ToStringEncoding = 'hex' | 'base64' | 'latin1';
+export type HashaInput = string | string[] | Buffer | Buffer[];
+export type HashaEncoding = ToStringEncoding | 'buffer';
+
+export interface Options<EncodingType = HashaEncoding> {
+	/**
+	 * Encoding of the returned hash.
+	 *
+	 * @default 'hex'
+	 */
+	readonly encoding?: EncodingType;
+
+	/**
+	 * Values: `md5` `sha1` `sha256` `sha512` *([Platform dependent](https://nodejs.org/api/crypto.html#crypto_crypto_createhash_algorithm))*
+	 *
+	 * *The `md5` algorithm is good for [file revving](https://github.com/sindresorhus/rev-hash), but you should never use `md5` or `sha1` for anything sensitive. [They're insecure.](http://googleonlinesecurity.blogspot.no/2014/09/gradually-sunsetting-sha-1.html)*
+	 *
+	 * @default 'sha512'
+	 */
+	readonly algorithm?: string;
```

Comment:
Can you list the encodings here? It says "platform dependent", but I'm pretty sure these algorithms are available everywhere.

Reply:
But should I still allow plain strings?

If so, I don't think that it makes a difference (IDE-wise) to list algorithms when allowing plain strings. There will be no auto-completion for the listed algorithms.

I could, of course do it anyway.

Reply:
Yes, it should still allow plain string. There's a workaround to get this working, see: https://github.com/sindresorhus/boxen/pull/39#discussion_r252933967

Reply:
Wow!